File: src/main/kotlin/cz/abo/b2b/web/shoppingcart/ShoppingCart.java
package cz.abo.b2b.web.shoppingcart;

import java.util.HashMap;

public class ShoppingCart extends HashMap<String, Long> {


    public void add(String productId, int count) {
        Long quantity = getOrDefault(productId, 0L);
        quantity += count;
        put(productId, quantity);
    }
}

File: src/main/kotlin/cz/abo/b2b/web/shoppingcart/ShoppingCart.kt
package cz.abo.b2b.web.shoppingcart

import cz.abo.b2b.web.dao.Product
import java.util.*
import kotlin.collections.HashMap

open class ShoppingCart : HashMap<UUID, ShoppingCartSupplier>() {
    fun add(product: Product, count: Long) {
        val shoppingCartSupplier = getOrDefault(product.supplier.id, ShoppingCartSupplier(product.supplier))
        put(product.supplier.id, shoppingCartSupplier)
        shoppingCartSupplier.addToCart(product, count)
    }
}